onnxruntime/wasm/api.h

```diff
@@ -222,4 +230,132 @@ int EMSCRIPTEN_KEEPALIVE OrtRun(ort_session_handle_t session,
+ * @return a handle of the ORT training session
+ *
+ */
+ort_training_session_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingCreateSession(const ort_session_options_handle_t session_options,
+                                                                            const ort_training_checkpoint_handle_t training_checkpoint_state_handle,
+                                                                            void* train_model,
+                                                                            size_t train_size,
+                                                                            void* eval_model,
+                                                                            size_t eval_size,
+                                                                            void* optimizer_model,
+                                                                            size_t optimizer_size);
+
+/**
+ * Resets the gradients of all trainable parameters to zero for the specified TrainingSession
+ * @param training_handle handle of the training session
+ * @returns ORT error code. If not zero, call OrtGetLastError() to get detailed error message.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtTrainingLazyResetGrad(ort_training_session_handle_t training_session);
+
+/**
+ * @brief Run a single training step.
+ *
+ * @param training_session_handle session handle of the specified session
+ * @param inputs user inputs to the training model
+ * @param inputs_count number of user inputs to the training model
+ * @param outputs user outputs computed by train step
+ * @param outputs_count number of user outputs expected from thsi train step
+ * @param run_options handle of the run options
+ * @return int ORT error code. If not zero, call OrtGetLastError() to get detailed error message.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtTrainingRunTrainStep(ort_training_session_handle_t training_session_handle,
+                                                 const ort_tensor_handle_t inputs, const size_t inputs_count,
+                                                 ort_tensor_handle_t outputs,
+                                                 const size_t outputs_count,
+                                                 OrtRunOptions* run_options = nullptr);
+
+/**
+ * Performs weight updates for the trainable parameters in the given training session using the optimizer model.
+ * @param training_handle handle of the training session
+ * @param run_options optional parameter of run options for this training step
+ * @returns ORT error code. If not zero, call OrtGetLastError() to get detailed error message.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtTrainingOptimizerStep(ort_training_session_handle_t training_handle,
+                                                  const ort_run_options_handle_t run_options = nullptr);
+
+ort_tensor_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingEvalStep(ort_training_session_handle_t training_handle,
```

Comment:
add document.

This function seems very similar to OrtTrainingRunTrainStep (they even have very similar implementation), but have very different function signature. is this intentional?

Reply:
Added documentation. & yes, a TrainingSession is created with a training model, and optionally, eval and optimizer models (total of 3 models). runTrainStep runs the step for the training model, while run EvalStep runs the step & computes outputs for the eval model.